actionpack/test/dispatch/server_timing_test.rb

```diff
@@ -36,49 +36,58 @@ def create
   test "server timing header is included in the response" do
     with_test_route_set do
       get "/"
-      assert_match(/\w+/, @response.headers["Server-Timing"])
+      assert_match(/\w+/, @response.headers["server-timing"])
     end
   end
 
   test "includes default action controller events duration" do
     with_test_route_set do
       get "/"
-      assert_match(/start_processing.action_controller;dur=\w+/, @response.headers["Server-Timing"])
-      assert_match(/process_action.action_controller;dur=\w+/, @response.headers["Server-Timing"])
+      assert_match(/start_processing.action_controller;dur=\w+/, @response.headers["server-timing"])
+      assert_match(/process_action.action_controller;dur=\w+/, @response.headers["server-timing"])
     end
   end
 
   test "includes custom active support events duration" do
     with_test_route_set do
       get "/id"
-      assert_match(/custom.event;dur=\w+/, @response.headers["Server-Timing"])
+      assert_match(/custom.event;dur=\w+/, @response.headers["server-timing"])
     end
   end
 
   test "events are tracked by thread" do
     barrier = Concurrent::CyclicBarrier.new(2)
-
-    stub_app = -> (env) {
-      env["proc"].call
-      [200, {}, "ok"]
-    }
-    app = ActionDispatch::ServerTiming.new(stub_app)
+    env = Rack::MockRequest.env_for("", {})
 
     t1 = Thread.new do
-      app.call({ "proc" => -> {
+      proc = -> (env) {
```

Comment:
> The tests that rely on a `Concurrent::CyclicBarrier` ("events are tracked by thread") were changed since passing the required proc in the env is not compatible with the SPEC:

You may be able to change the header to `action_dispatch.test` to get around this instead of modifying how the test works,

Reply:
Nice idea. Done!